Packs/Netmiko/Integrations/Netmiko/Netmiko.py

```diff
@@ -0,0 +1,164 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+''' IMPORTS '''
+
+
+import io
+import json
+import re
+import sys
+from datetime import datetime
+from time import sleep
+
+from netmiko import Netmiko
+from netmiko.ssh_dispatcher import platforms
+from paramiko import PKey
+
+''' HELPER FUNCTIONS '''
+
+
+class Client:
+    def __init__(self, platform, hostname, username, password, port, ssh_key=None):
+        self.platform = platform
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.port = port
+        self.ssh_key = ssh_key
+        self.net_connect = None
+
+    def connect(self):
+        try:
+            if self.ssh_key:
+                with open("ssh_key", "w") as fp:
+                    fp.write(self.ssh_key)
+                self.net_connect = Netmiko(device_type=self.platform, host=self.hostname, port=self.port,
+                                           use_keys=True, key_file="ssh_key", username=self.username, passphrase=self.password)
+            else:
+                self.net_connect = Netmiko(device_type=self.platform, host=self.hostname,
+                                           port=self.port, username=self.username, password=self.password)
+        except Exception as err:
+            return_error(err)
+
+    def disconnect(self):
+        try:
+            if self.net_connect:
+                self.net_connect.disconnect()
+        except Exception as err:
+            return_error(err)
+
+    def cmds(self, exitRequired, exitChar, commands, enable, isConfig):
+        try:
+            output = {"Hostname": self.hostname, "Platform": self.platform, "Commands": []}
+            self.connect()
+            sleep(2)
+            if enable:
+                self.net_connect.enable()
+            if isConfig:
+                output['Commands'].append({"Hostname": self.hostname, "DateTime(UTC)": datetime.utcnow(
+                ).isoformat(), "Config": self.net_connect.send_config_set(commands)})
+            if not isConfig:
+                for cmd in commands:
+                    prompt = self.net_connect.find_prompt()
+                    c = {"Hostname": self.hostname, "DateTime(UTC)": datetime.utcnow().isoformat(
+                    ), "Command": cmd, "Output": f"{prompt} {self.net_connect.send_command_timing(cmd)}"}
+                    output['Commands'].append(c)
+
+        except Exception as err:
+            return_error(err)
+        finally:
+            self.disconnect()
+        return output
+
+
+def test_command(client):
+    client.connect()
+    client.disconnect()
+    demisto.results('ok')
+    sys.exit(0)
+
+
+def cmds_command(client):
```

Comment:
It is better to avoid calling `demisto.args()` and `demisto.params()` multiple times.